domain/application/state/application_test.go

```diff
@@ -980,6 +980,75 @@ func (s *applicationStateSuite) TestUpsertCloudServiceAnother(c *gc.C) {
 	c.Assert(providerIds, jc.SameContents, []string{"provider-id", "another-provider-id"})
 }
 
+func (s *applicationStateSuite) TestUpsertCloudServiceUpdateRemoveAddresses(c *gc.C) {
+	appID := s.createApplication(c, "foo", life.Alive)
+	s.createApplication(c, "bar", life.Alive)
+	err := s.state.UpsertCloudService(context.Background(), "foo", "provider-id", network.SpaceAddresses{
+		network.SpaceAddress{
+			MachineAddress: network.MachineAddress{
+				Value:      "10.0.0.1",
+				ConfigType: network.ConfigStatic,
+				Type:       network.IPv4Address,
+				Scope:      network.ScopeFanLocal,
```

Comment:
I wouldn't use this scope, even in tests. We'll want to delete it from the code base at some point.

Reply:
Fixed.